scripts/ci/create_workspace.py

```diff
@@ -20,27 +20,57 @@
 from urllib.request import urlretrieve
 
 from ros_buildfarm.argument import add_argument_repos_file_urls
+from ros_buildfarm.argument import add_argument_repository_names
+from ros_buildfarm.argument import add_argument_rosdistro_name
 from ros_buildfarm.argument import add_argument_test_branch
 from ros_buildfarm.common import Scope
 from ros_buildfarm.vcs import export_repositories, import_repositories
 from ros_buildfarm.workspace import ensure_workspace_exists
+from rosdistro import get_distribution
+from rosdistro import get_index
+from rosdistro import get_index_url
+import yaml
 
 
 def main(argv=sys.argv[1:]):
     parser = argparse.ArgumentParser(
         description='Create a workspace from vcs repos files.')
-    add_argument_repos_file_urls(parser, required=True)
+    add_argument_rosdistro_name(parser)
+    add_argument_repos_file_urls(parser)
+    add_argument_repository_names(parser)
     add_argument_test_branch(parser)
     parser.add_argument(
         '--workspace-root',
         help='The path of the desired workspace',
         required=True)
     args = parser.parse_args(argv)
 
+    assert args.repos_file_urls or args.repository_names
+
     ensure_workspace_exists(args.workspace_root)
 
+    repos_files = []
+    if args.repository_names:
+        with Scope('SUBSECTION', 'get repository information from rosdistro'):
+            index = get_index(get_index_url())
+            dist = get_distribution(index, args.rosdistro_name)
```

Comment:
How will this act if `repository_names` is specified but `rosdistro_name` is not?

Reply:
`rosdistro_name` is a required positional argument. So it must be / is always set.

Reply:
Understood.